src/librbd/AioObjectRequest.cc

```diff
@@ -410,27 +451,10 @@ namespace librbd {
           		       << m_object_off << "~" << m_object_len << dendl;
         m_state = LIBRBD_AIO_WRITE_PRE;
 
-        uint8_t new_state;
```

Comment:
That's what I am still missing I guess -- the PRE state was just around for updating the object map.  The only difference that I see is that instead of calling ```send_write``` it calls ```send_write_op```?